client/src/test/java/org/opensearch/geospatial/action/IpEnrichmentActionClientTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.action;
+
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.action.ActionFuture;
+import org.opensearch.core.action.ActionResponse;
+
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class IpEnrichmentActionClientTest {
```

Comment:
For all the tests lets use the same test framework used in plugin rather than using standard Mockito ensure that test framework is same across different modules of the plugin.

Reply:
I have updated the build.gradle to read the Mockito version from the `opensearch.build` plugin, in this case we will have the same version of Mockito as the main OpenSearch repo.
```
testImplementation "org.mockito:mockito-core:${versions.mockito}"
```
Or you mean I should ditch Mockito all together?

Reply:
It seems we don't extend from `OpenSearchTestCase` for client module? https://github.com/opensearch-project/ml-commons/blob/main/client/src/test/java/org/opensearch/ml/client/MachineLearningNodeClientTest.java

Reply:
The problem with not doing this would be that we first we will not able to test the code with random data and another problem is since we will not have seed whenever we want to reproduce a failure. Not sure why MLCommons didn't do this. This is an acceptable practice in Opensearch Plugins. Plus you will miss out on all the different mocking and testing framework added by Opensearch.

Reply:
In my point, this depends on the amount of logic to test. 
I agree we should have it to remain the consistency, but for now if we aim to have it as a thin client which has no actual logic inside except the argument validation.
This may be a overkill.